include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,425 @@
+    //! Number of macro particles
+    std::unique_ptr< IntBuffer > nmp;
+
+    MappingDesc *cellDescription;
+    std::string notifyPeriod;
+    std::string speciesName;
+    std::string pluginName;
+    std::string pluginPrefix;
+    std::string filename_prefix;
+
+    /** Range of scattering vector
+     * The scattering vecotor here is defined as
+     * 4*pi*sin(theta)/lambda, where 2theta is the angle
+     * between scattered and incident beam
+     **/
+    float3_X q_min, q_max, q_step;
+    //! Number of scattering vectors
+    DataSpace< 3 > numVectors;
+
+    std::vector< float_X > sumfcoskr_master;
+    std::vector< float_X > sumfsinkr_master;
+    std::vector< float_X > intensity_master;
+    float_X np_master;
+    int64_t nmp_master;
+
+    bool isMaster;
+
+    uint32_t currentStep;
+
+    mpi::MPIReduce reduce;
+
+  public:
+    Saxs()
+        : pluginName(
+              "SAXS: calculate the SAXS scattering intensity of a species"),
+          speciesName(ParticlesType::FrameType::getName()),
+          pluginPrefix(speciesName + std::string("_saxs")),
+          filename_prefix(pluginPrefix), cellDescription(nullptr), isMaster(false),
+          currentStep(0)
+    {
+        Environment<>::get().PluginConnector().registerPlugin(this);
+    }
+
+    virtual ~Saxs() {}
+
+    /**
+     * This function represents what is actually calculated if the plugin
+     * is called. Here, one only sets the particles pointer to the data of
+     * the latest time step and calls the 'calculateSAXS'
+     * function if for the actual time step radiation is to be calculated.
+     * @param currentStep
+     */
+    void notify(uint32_t currentStep)
+    {
+        std::cout << "SAXS plugin enabled" << std::endl;
+        calculateSAXS(currentStep);
+    }
+
+    void pluginRegisterHelp(po::options_description &desc)
+    {
+        desc.add_options()((pluginPrefix + ".period").c_str(),
+            po::value<std::string>(&notifyPeriod),
+            "enable plugin [for each n-th step]")(
+            (pluginPrefix + ".qx_max").c_str(),
+            po::value<float_X>(&q_max[0])->default_value(5),
+            "reciprocal space range qx_max (A^-1)")(
+            (pluginPrefix + ".qy_max").c_str(),
+            po::value<float_X>(&q_max[1])->default_value(5),
+            "reciprocal space range qy_max (A^-1)")(
+            (pluginPrefix + ".qz_max").c_str(),
+            po::value<float_X>(&q_max[2])->default_value(5),
+            "reciprocal space range qz_max (A^-1)")(
+            (pluginPrefix + ".qx_min").c_str(),
+            po::value<float_X>(&q_min[0])->default_value(-5),
+            "reciprocal space range qx_min (A^-1)")(
+            (pluginPrefix + ".qy_min").c_str(),
+            po::value<float_X>(&q_min[1])->default_value(-5),
+            "reciprocal space range qy_min (A^-1)")(
+            (pluginPrefix + ".qz_min").c_str(),
+            po::value<float_X>(&q_min[2])->default_value(-5),
```

Comment:
The default value for `numVectors[2]` is 1. Since
https://github.com/ComputationalRadiationPhysics/picongpu/blob/866019f9df9d5f62f47a92dbe35949503813f1ea/include/picongpu/plugins/saxs/Saxs.kernel#L238 and
https://github.com/ComputationalRadiationPhysics/picongpu/blob/866019f9df9d5f62f47a92dbe35949503813f1ea/include/picongpu/plugins/saxs/Saxs.kernel#L244 scattering is only calculated for q_z equal -5 A^-1 but I think that, for SAXS, q_z = 0 would make more sense (since it's a small angle).  
```suggestion
            po::value<float_X>(&q_min[2])->default_value(0),
```
This matters for 3D simulations where r_z is not 0.

Reply:
@pordyna thanks, good point.
Somewhat related: with this notation, the `q_max` value is never reached, so with the natural `q_min = -q_max` it would be asymmetrical (slightly if the number of vectors is large).